### ISSUE TITLE
Gender Inputs

### ISSUE THREAD

**Issue by innat** [chenchao666/Bone-Age-Assessment#3]
What are you doing here?

https://github.com/chenchao666/Bone-Age-Assessment/blob/31e23ff5638c88ed9f89035d921be9782b693a69/main_classification.py#L40

The gender should be one feature vector with `1` for male and `0` for female, isn't it? It causes gradients to exploit or vanishing gradients in training time for such inputs.

**Comment by chenchao666** [comment]
> What are you doing here?
> 
> https://github.com/chenchao666/Bone-Age-Assessment/blob/31e23ff5638c88ed9f89035d921be9782b693a69/main_classification.py#L40
> 
> The gender should be one feature vector with `1` for male and `0` for female, isn't it? It causes gradients to exploit or vanishing gradients in training time for such inputs.

I convert it to 1 and -1 for male and feale, why it causes the gradient explosion or vanishing ？ That works in my code. and some other work for BAA also have similar process

**Comment by innat** [comment]
@chenchao666 
Would you please check [this ](https://www.kaggle.com/ipythonx/notebook1c4c4cb749/notebook?scriptVersionId=56426360)notebook? There probably some silly mistake that causes `inf` or `nan` loss sometimes. It's the same task, same data set. But I didn't use image data but the gender input.

**Comment by chenchao666** [comment]
> @chenchao666
> Would you please check [this ](https://www.kaggle.com/ipythonx/notebook1c4c4cb749/notebook?scriptVersionId=56426360)notebook? There probably some silly mistake that causes `inf` or `nan` loss sometimes. It's the same task, same data set. But I didn't use image data but the gender input.

Do you mean you only utlize the gender as input ? That is completely wrong. The model will never converge with only gender information as input, since your model has different outputs for the same input.

**Comment by innat** [comment]
OK. Let me explain. Initially, I did make two inputs, one is the image and another one is gender. But suddenly loss went `inf` or `nan` type. Then I did input only image data but not gender information and at that time no such issue. And next, I input the only gender but not the image data, and this issue came. So, my primary assumption was there probably some issue with the gender input. So, the above notebook only uses gender information.
